src/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -122,6 +124,96 @@ public static void JsonIgnoreAttribute()
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<WrapperForClassWithUnsupportedDictionary>(wrapperJson));
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Serialize(new WrapperForClassWithUnsupportedDictionary()));
+
+            // When ignored, we can serialize and deserialize without exceptions.
+            ClassWithIgnoredUnsupportedDictionary obj = JsonSerializer.Deserialize<ClassWithIgnoredUnsupportedDictionary>(json);
+            Assert.Null(obj.MyDict);
+            Assert.Equal("{}", JsonSerializer.Serialize(new ClassWithIgnoredUnsupportedDictionary()));
+
+            WrapperForClassWithIgnoredUnsupportedDictionary wrapperObj = JsonSerializer.Deserialize<WrapperForClassWithIgnoredUnsupportedDictionary>(wrapperJson);
+            Assert.Null(wrapperObj.MyClass.MyDict);
+            Assert.Equal(@"{""MyClass"":{}}", JsonSerializer.Serialize(new WrapperForClassWithIgnoredUnsupportedDictionary()
+            {
+                MyClass = new ClassWithIgnoredUnsupportedDictionary(),
+            })); ;
+        }
+
+        [Fact]
+        public static void JsonIgnoreAttribute_UnsupportedBigInteger()
+        {
+            string json = @"{""MyBigInteger"":1}";
+            string wrapperJson = @"{""MyClass"":{""MyBigInteger"":1}}";
+
+            // Unsupported types will throw by default.
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<ClassWithUnsupportedBigInteger>(json));
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<WrapperForClassWithUnsupportedBigInteger>(wrapperJson));
+
+            // When ignored, we can serialize and deserialize without exceptions.
+            ClassWithIgnoredUnsupportedBigInteger obj = JsonSerializer.Deserialize<ClassWithIgnoredUnsupportedBigInteger>(json);
+            Assert.Null(obj.MyBigInteger);
+            Assert.Equal("{}", JsonSerializer.Serialize(new ClassWithIgnoredUnsupportedBigInteger()));
+
+            WrapperForClassWithIgnoredUnsupportedBigInteger wrapperObj = JsonSerializer.Deserialize<WrapperForClassWithIgnoredUnsupportedBigInteger>(wrapperJson);
+            Assert.Null(wrapperObj.MyClass.MyBigInteger);
+            Assert.Equal(@"{""MyClass"":{}}", JsonSerializer.Serialize(new WrapperForClassWithIgnoredUnsupportedBigInteger()
+            {
+                MyClass = new ClassWithIgnoredUnsupportedBigInteger(),
+            })); ;
+        }
+
+        public class ClassWithUnsupportedDictionary
+        {
+            public ConcurrentDictionary<object, object> MyDict { get; set; }
+        }
+
+        public class WrapperForClassWithUnsupportedDictionary
+        {
+            public ClassWithUnsupportedDictionary MyClass { get; set; } = new ClassWithUnsupportedDictionary();
+        }
+
+        public class ClassWithIgnoredUnsupportedDictionary
+        {
+            [JsonIgnore]
+            public ConcurrentDictionary<object, object> MyDict { get; set; }
```

Comment:
Can you add other non-supported collection/dictionaries as well?

`IDict<object, object>`, `Dict<int, string>`, other collections, etc.